src/Widgets/AbstractAppContainer.vala

```diff
@@ -80,6 +82,19 @@ namespace AppCenter {
             }
         }
 
+        private string? suggested_amount_ = null;
+        public string suggested_amount {
+            get {
+                if (this.package == null || this.package.component == null || updates_view) {
+                    return DEFAULT_PRICE_DOLLARS;
+                } else if (suggested_amount_ == null) {
+                    suggested_amount_ = this.package.component.get_custom_value ("x-appcenter-suggested-price");
+                }
+
+                return suggested_amount_ != null ? suggested_amount_ : DEFAULT_PRICE_DOLLARS;
+            }
+        }
+
```

Comment:
@Philip-Scott I've taken a look at this, and it seems as though the package metadata is not available at the point we construct the button, so the code makes sense as it is at the moment. The button is already set to free when no stripe key is available by the `action_button.can_purchase = payments_enabled;` line in the same place as setting the amount.